python/ray/experimental/state.py

```diff
@@ -304,3 +305,24 @@ def client_table(self):
       node_info[node_ip_address].append(client_info_parsed)
 
     return node_info
+
+  def task_profiles(self):
+    """Fetch and return a list of task profiles.
+    Returns:
+      A list of task profiles.
+    """
+    event_names = self.redis_client.keys("event_log*")
+    results = dict()
+    for i in range(len(event_names)):
+      event_list = self.redis_client.lrange(event_names[i], 0, -1)
+      for event in event_list:
+        event_dict = json.loads(event.decode("ascii"))
+        task_id = ""
+        for element in event_dict:
+          if 'task_id' in element[3]:
+            task_id = element[3]['task_id']
+        if task_id == "":
+          return "Task_id not found. This shouldn't happen."
```

Comment:
We probably don't want to return this string, perhaps in general we should just return a tuple of two things, the first is a dictionary mapping task IDs to the corresponding event, and the second is a list of the events that don't have task IDs.